src/skottie-player.ts

```diff
@@ -554,14 +565,163 @@ export class SkottiePlayer extends LitElement {
     const file = files?.[0];
     if (file && mapKey) {
       const reader = new FileReader();
-      reader.readAsArrayBuffer(file);
-      reader.onload = () => {
-        (this.assets as any)[mapKey] = reader.result;
-        this.updatePlayer();
+      if (mapKey === 'Image_0.jpeg') {
+        this.updateColors(reader, file, mapKey);
+      } else {
+        this.readAssetFile(reader, file, mapKey);
+      }
+    }
+  }
+
+  /**
+   * Updates template colors based on an uploaded image. Also reads in the
+   * uploaded image as an image asset.
+   * 
+   * @param reader  Used when reading image as asset
+   * @param file    Image file
+   * @param mapKey  Name of asset, should always be "Image_0.jpeg"
```

Comment:
should always be "Image_0.jpeg" -> this comment might be too strict, as asset names may vary in different templates.